src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -2008,15 +2008,41 @@
  * @since 1.5
  */
 public final class Formatter implements Closeable, Flushable {
+    // Caching DecimalFormatSymbols. Non-volatile to avoid thread slamming.
+    private static DecimalFormatSymbols DFS = null;
+    private static DecimalFormatSymbols getDecimalFormatSymbols(Locale locale) {
+        // Capture local copy to avoid thread race.
+        DecimalFormatSymbols dfs = DFS;
+        if (dfs != null && dfs.getLocale().equals(locale)) {
+            return dfs;
+        }
+        // Fetch a new local instance of DecimalFormatSymbols. Note that DFS are mutatble
```

Comment:
Thank you Jim for these comments about multi-threaded access. Looks good to me.
One minor typo on this line - "mutatble" should have been "mutable".

Reply:
Noted